### PR TITLE
Milestone: Testing

### PR DESCRIPTION
**Pull request by Lancelot-SO**
### The following changes were made:
- Added test for food model
-

### CODE REVIEW

File: spec/features/food_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                            description: 'Test description', public: true, user_id: @user.id)
+    @food = Food.create(name: 'Test food', price: 12.2, quantity: 4, measurement_unit: 'gram', user_id: @user.id)
+    @food_two = Food.create(name: 'Test food Two', price: 12.2, quantity: 4, measurement_unit: 'gram',
+                            user_id: @user.id)
+    @recipe_food = RecipeFood.create(quantity: 10, recipe_id: @recipe.id, food_id: @food.id)
+    visit new_user_session_path
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Log in'
+  end
```
Comment (ram1117) [inline spec/features/food_spec.rb:17]: Hi @Lancelot-SO if this is integration test, you have to create assertions to check if web page displays everything as expected. You have to write tests for Food items index page, food item new page etc. You also have to write test cases for Show public recipes index page

File: spec/models/food_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    expect(@food).to_not be_valid
+  end
+  it 'is not valid without a user id' do
+    @food.user_id = nil
+    expect(@food).to_not be_valid
+  end
+  it 'is not valid without a unique name' do
+    @food.name = @food_two.name
+    expect(@food).to be_valid
+  end
```
Comment (ram1117) [inline spec/models/food_spec.rb:24]: @Lancelot-SO you have to write test cases for validations in the Food model. You wrote test cases only for id and name attributes. you have to write test cases for

validates :measurement_unit, presence: true
  validates :price, numericality: { greater_than_or_equal_to: 0 }
  validates :quantity, numericality: { only_integer: true, greater_than_or_equal_to: 0 }

**Review by ram1117 — CHANGES_REQUESTED** [review]
Hi @Lancelot-SO Please see the comments

File: spec/features/food_spec.rb (hunk 1 above)
Comment (Lancelot-SO) [reply spec/features/food_spec.rb:17]: I am creating for the other model files too. i just pushed it because I want to do it one after the other

Comment (Lancelot-SO) [reply spec/features/food_spec.rb:17]: I have added test for the other models hope you check them and review them.

**Comment by ram1117** [comment]
Hi @Lancelot-SO some test cases are still failing

image

**Review by ram1117 — APPROVED** [review]

**Pull request merged**